bundles/paikkatietoikkuna/mapmodule/mapmodule.olcs.js

```diff
@@ -454,6 +456,122 @@ class MapModuleOlCesium extends MapModuleOl {
         return state;
     }
 
+    /**
+     *
+     * @method centerMap
+     * Moves the map to the given position and zoomlevel. Overrides 2d centerMap function.
+     * @param {Number[] | Object} lonlat coordinates to move the map to
+     * @param {Number/OpenLayers.Bounds/Object} zoomLevel zoomlevel to set the map to
+     * @param {Boolean} suppressEnd true to NOT send an event about the map move
+     *  (other components wont know that the map has moved, only use when chaining moves and
+     *     wanting to notify at end of the chain for performance reasons or similar) (optional)
+     * @param {Object} options  has values for heading, pitch, roll and duration
+     */
+    centerMap (lonlat, zoom, suppressEnd, options) {
+        lonlat = this.normalizeLonLat(lonlat);
+        const location = olProj.transform([lonlat.lon, lonlat.lat], this.getProjection(), 'EPSG:4326');
+        const cameraHeight = this.adjustZoom(zoom);
+        const duration = options && options.duration ? options.duration : 3000;
+        const animationDuration = duration / 1000;
+        const camera = options && options.heading && options.roll && options.pitch
+            ? { heading: options.heading,
+                roll: options.roll,
+                pitch: options.pitch } : undefined;
+
+        if (options && options.animation) {
+            this._flyTo(location[0], location[1], cameraHeight, animationDuration, camera);
+            if (suppressEnd !== true) {
+                setTimeout(() => {
+                    this.notifyMoveEnd();
```

Comment:
Please check if the AfterMapMoveEvent is triggered on 3D without this (as it was on 2D)?

Reply:
It is triggered also, so removing this